src/Farmer/Builders/Builders.WebApp.fs

```diff
@@ -478,6 +479,43 @@ type WebAppConfig =
                 for (_,slot) in this.CommonWebConfig.Slots |> Map.toSeq do
                     slot.ToArm site
 
+            match this.CustomDomain with
+            | SecuredDomain (customDomain, certOptions) ->
+                let hostNameBinding =
+                    { Location = location
+                      SiteId =  Managed (Arm.Web.sites.resourceId this.Name)
+                      DomainName = customDomain
+                      SslState = SslDisabled }
+                let cert =
+                    { Location = location
+                      SiteId = this.ResourceId
+                      ServicePlanId = this.ServicePlanId
+                      DomainName = customDomain }
+                hostNameBinding
+                cert
+                let resourceLocation = location
+                yield! (resourceGroup { name (ArmExpression.create "resourceGroup().name")
+                                        location resourceLocation
+                                        add_resource { hostNameBinding with
+                                                        SslState = match certOptions with
+                                                                   | None -> Sni (ArmExpression.reference(Arm.Web.certificates, Arm.Web.certificates.resourceId $"{this.Name.Value}-cert").Map(sprintf "%s.Thumbprint"))
```

Comment:
Shouldn't this be `SslState.Disabled`? They've said they want a custom domain with no cert? Do we need cert options to be and `Option<>` at all here seems like it should always be required for a Secured Domain?